src/ast/rewriter/seq_rewriter.cpp

```diff
@@ -3775,8 +4148,10 @@ br_status seq_rewriter::mk_str_in_regexp(expr* a, expr* b, expr_ref& result) {
 
     expr_ref hd(m()), tl(m());
     if (get_head_tail(a, hd, tl)) {
-        result = re().mk_in_re(tl, re().mk_derivative(hd, b));
-        return BR_REWRITE2;
+        //result = re().mk_in_re(tl, re().mk_derivative(hd, b));
+        //result = re().mk_in_re(tl, mk_derivative(hd, b));
+        result = mk_in_antimirov(tl, mk_antimirov_deriv(hd, b, m().mk_true()));
+        return BR_DONE;
```

Comment:
BR_DONE is incorrect. There could be more rewrites. z3test/regressions/smt2/re.smt2 regresses. It is fixed by returning BR_REWRITE1

Reply:
fixed